Documentation/operations/troubleshooting_clustermesh.rst

```diff
@@ -171,6 +171,24 @@ you may perform the following steps to troubleshoot ClusterMesh issues.
     * A firewall between the local cluster and the remote cluster may drop the
       control plane connection. Ensure that port 2379/TCP is allowed.
 
+    * The connection may fail to establish when all the following conditions
+      are met:
+
+      * KVStoreMesh is disabled,
+      * The Host Firewall is enabled,
+      * tunneling is enabled,
+      * Cilium's kube-proxy-replacement is enabled,
```

Comment:
I managed to reproduce a sort of similar issue even with KPR disabled, if kvstoremesh is enabled (i.e., traffic is pod to nodeport, rather than node to nodeport). My guess is that the latter case is instead handled by [this logic](https://github.com/cilium/cilium/blob/43642e5cda6eeb8c7ab7af9d64579ce725501e85/bpf/bpf_lxc.c#L1209-L1211), which forwards the nodeport reply to the stack, and then back via the native device.

Let's consider a scenario in which cluster A connected to cluster B, but not vice versa (e.g., commenting out the last step from your patch). The initial connection from kvstoremesh on A to the clustermesh-apiserver on B works because neither hosting agents knows about the nodes in the other cluster. However, after the initial connection, A learns about the nodes in B, and subsequent connections to the clustermesh-apiserver on B will go through the tunnel, and get dropped on receive side due to the missing wireguard keys. 

This can be verified running `kubectl exec -t -n kube-system deploy/clustermesh-apiserver -c kvstoremesh -- clustermesh-apiserver kvstoremesh-dbg troubleshoot` on A, which fails, while `...  kvstoremesh-dbg status` reports success.

Although the situation can converge if B connects back to A before that the initial TCP connection gets closed, this is still risky. Hence, I'd suggest to update the docs to not mention KPR as one of the requisites for this issue to happen, considering that even KPR disabled can still trigger issues in certain conditions.

Reply:
Nice find! OK, I'll drop the KPR “prerequisite” to trigger the issue (and from the Helm validation).

I'll also report your comment to the original issue, so we can better remember it.